packages/hash/frontend/src/blocks/page/BlockConfigMenu/BlockConfigMenu.tsx

```diff
@@ -0,0 +1,195 @@
+              checked={typeof value === "boolean" ? value : false}
+            />
+          }
+          label={name}
+        />
+      );
+
+    case "string":
+      if (format) {
+        // @todo validate string format - should have a reusable input that does this
+      }
+      return (
+        <TextField
+          label={name}
+          onBlur={enumList ? undefined : updateProperty}
+          onChange={(event) => {
+            setDraftValue(event.target.value);
+            if (enumList) {
+              updateProperty(event);
+            }
+          }}
+          select={!!enumList}
+          value={draftValue}
+          variant="outlined"
+        >
+          {(enumList ?? []).map((option: string) => (
+            <MenuItem key={option} value={option}>
+              {option}
+            </MenuItem>
+          ))}
+        </TextField>
+      );
+
+    case "number":
+      return (
+        <TextField
+          defaultValue={value ?? ""}
+          onBlur={(event) => onChange(event.target.value)}
+          onChange={(event) => setDraftValue(event.target.value)}
+          type="number"
+          value={draftValue}
+          variant="outlined"
+        />
+      );
+
+    default:
+      throw new Error(`Property type ${type} config input not implemented`);
+  }
+};
+
+type BlockConfigMenuProps = {
+  anchorRef: ForwardedRef<HTMLDivElement>;
+  blockData: BlockEntity | null;
+  blockSchema?: JsonSchema | null;
+  closeMenu: () => void;
+  popupState: PopupState;
+  updateConfig: (propertiesToUpdate: JSONObject) => void;
+};
+
+/**
+ * Provides a generic UI for setting properties on the entity a block is rendering.
+ * Useful for when the block doesn't provide the UI to do so itself.
+ * Only the properties listed in the block's schema as 'configProperties' are shown.
+ */
+export const BlockConfigMenu: VoidFunctionComponent<BlockConfigMenuProps> = ({
+  anchorRef,
+  blockData,
+  blockSchema,
+  closeMenu,
+  popupState,
+  updateConfig,
+}) => {
+  useKey(["Escape"], closeMenu);
+
+  const configProperties = extractConfigPropertySchemas(blockSchema ?? {});
+
+  const entityData = blockData?.properties.entity.properties as
+    | JSONObject
+    | undefined;
+
+  if (anchorRef && typeof anchorRef === "function") {
+    throw new Error(
+      "BlockConfigMenu requires an element attached to anchorRef.current, to anchor the menu to.",
+    );
+  }
+
+  return (
+    <Popover
+      {...bindPopover(popupState)}
+      anchorEl={anchorRef?.current}
+      PaperProps={{ sx: { padding: 2 } }}
+    >
+      <Typography variant="h5">Configure block</Typography>
+      {configProperties.map(([name, schema]) => (
+        <Box key={name} sx={{ mt: 2 }}>
+          <ConfigurationInput
+            name={name}
+            onChange={(value: any) => updateConfig({ [name]: value })}
+            rootSchema={blockSchema ?? {}}
+            propertySchema={schema}
+            value={entityData?.[name]}
```

Comment:
This brings up a warning on the select component if the value is undefined. 

![Screenshot 2022-04-25 at 00 39 16](https://user-images.githubusercontent.com/16928688/165001586-15f50e08-2829-48b5-9e91-ee5c35119cb7.png)

Can be fixed by doing

```suggestion
            value={entityData?.[name] ?? ""}
```

Reply:
Thank you - https://github.com/hashintel/hash/pull/490/commits/c17be4e9e3738af9d81cf5b52ce2816f77672426